components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -429,6 +449,100 @@ public String exportApplication(String applicationId, Boolean exportSecrets) {
         }
     }
 
+    /**
+     * Export an application identified by the applicationId, in the given format.
+     *
+     * @param fileType      The format of the exported string.
+     * @param applicationId ID of the application to be exported.
+     * @param exportSecrets If True, all hashed or encrypted secrets will also be exported.
+     * @return string of the application in the given format.
+     */
+    public TransferResource exportApplicationAsFile(String fileType, String applicationId, Boolean exportSecrets) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Exporting application: " + applicationId + " as a file of type: " + fileType);
+        }
+
+        if (StringUtils.isBlank(fileType)) {
+            throw new UnsupportedOperationException("No valid media type found");
+        }
+
+        ServiceProvider serviceProvider;
+        try {
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            serviceProvider = getApplicationManagementService().exportSPFromAppID(
+                    applicationId, exportSecrets, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error exporting application with id: " + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Application: " + applicationId + " exported successfully.");
```

Comment:
This would imply the export flow is completed at this point. Let us add this particular msg once the flow is completed and make sure the same debug log(same information) is not repeated during the flow. Recheck other places with debug logs.

Reply:
Refactored all debug logs to a minimal